src/plugins/intel_cpu/src/nodes/kernels/jit_kernel_base.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+namespace x64 = dnnl::impl::cpu::x64;
+
+struct jit_kernel_args_stub {};
+
+template<typename jit_params, typename jit_call_args>
```

Comment:
Do I correctly understand that those template parameters are introduced to enforce some static interface such us the constructor with `jit_params` and the functor operators with `jit_call_args`? If I'm right, I would like to propose the following. On the base level we need some base type like `jit_kernel_base` where we place all the common stuff, but without the template parameters, to reduce the amount of generated code. Under the "common stuff" I mean our beloved uni_* mnemonics, which we definitely should move from oneDNN to our scope to minimize our changes to oneDNN lib, also we need common parts for working with the register pool and the dynamic register management, code emitters management, etc. There could be some helper utilities for more convenient register pool usage, e.g. factory methods for register to hide register pool usage complexity. On the next level we can bring this templated static interface to enforce developers to stick with the basic `jit_kernel` design.

Reply:
@maxnick 
> ... On the base level we need some base type like `jit_kernel_base` where we place all the common stuff, but without the template parameters, to reduce the amount of generated code. ...

What is the problem with generated code ?
As for me developer will either user custom class:
```cpp
class jit_refinment_anchor_kernel_base {
    ...
};
```
Or will use common class:
```cpp
using jit_refinment_anchor_kernel_base = jit_kernel_base<jit_refinment_anchor_args, jit_refinment_anchor_call_args>;
```

For me it is the same amount of generated code by compiler, but with a templated base class we have common class that will allow to use `Template Method Pattern`:
```cpp
template<typename jit_params, typename jit_call_args>
class jit_kernel_base : public x64::jit_generator {
 public:
    using kernel_func = void (*)(const jit_call_args *);

    explicit jit_kernel_base(x64::cpu_isa_t max_cpu_isa, const jit_params &jqp)
        : x64::jit_generator{nullptr, 256 * 1024, true, max_cpu_isa}, kernel_{nullptr}, params_{jqp} {}
    virtual ~jit_kernel_base() {}

    status_t create_kernel() override {
        const status_t code = jit_generator::create_kernel();
        if (code != dnnl::impl::status::success) {
            IE_THROW() << "Could not create kernel. Error code: " << std::to_string(code) << ". " <<
                          "Xbyak error code: " << Xbyak::ConvertErrorToString(Xbyak::GetError());
        }
        kernel_ = reinterpret_cast<decltype(kernel_)>(jit_ker());
        return code;
    }

    void operator()(const jit_call_args* args) const {
        assert(kernel_);
        kernel_(args);
    }

    void operator()(const jit_call_args& args) const {
        this->operator()(&args);
    }

    void uni_<some_operation>(...) {
       // using RegistersPool and maybe StackAllocator
    }

    // This function is done as template method pattern
    void generate() {
        this->preambule();
        register_poll_ = std::make_shared<RegistersPool>(...);
        stack_allocator_ = std::make_shared<StackAllocator>(...);
        generate(registersPool, stackAllocator);
        register_poll_.reset();
        stack_allocator_.reset();
        this->postambule();
    }

 protected:
    jit_params params_;
    std::shared_ptr<RegistersPool> register_poll_;
    std::shared_ptr<StackAllocator> stack_allocator_;

 private:
    kernel_func kernel_;
};
```

Reply:
Yes. Sorry if my explanations are not clear. But that is what I said
1. >..On the base level we need some base type like jit_kernel_base where we place all the common stuff, but without the template parameters...
2. >"...On the next level we can bring this templated static interface to enforce developers to stick with the basic jit_kernel design..."
So yes, this two level base classes structure is what we need.

Reply:
Updated PR, you can see the whole concept that I have in mind